### ISSUE TITLE
reportImplicitStringConcatenation doesn't work with parenthesis

### ISSUE THREAD

**Issue by npip99** [DetachHead/basedpyright#892]
Simple example:

```python
good = "abc" "def" # Error: reportImplicitStringConcatenation
bad1 = ( "abc" "def" ) # No Error
bad2 = (
    "abc" # No Error
    "def"
)
```

**Comment by beauxq** [comment]
some context: https://github.com/microsoft/pyright/issues/2781

**Comment by DetachHead** [comment]
the example in that issue still doesn't work. was that change reverted later on or something?

**Comment by DetachHead** [comment]
my bad i misread the issue. so the current behavior is intentional, i guess we could add an option to enable the old behavior

**Comment by beauxq** [comment]
I think I would find it most useful to give an error report only if it doesn't report some other type error both with and without a comma.

```python
print("a" "b")  # no type error either with or without comma, so unclear if it's intentional

int("555555555555555555555555555"   # Would give a type error with a comma,
    "555555555555555555555555555")  # so without comma is probably intentional
```
But that seems difficult to implement. ("Hypothetically, would this report a type error if the code were different?")

**Comment by DetachHead** [comment]
yea that doesnt sound feasible imo. i think just having an option for the old behavior is sufficient
